src/EntryPoints/WikibaseFacetedSearchHooks.php

```diff
@@ -67,18 +71,39 @@ public static function onSpecialSearchResultsPrepend(
 		);
 	}
 
+	public static function onSpecialSearchResults(
+		string $term,
+		?ISearchResultSet $titleMatches,
+		?ISearchResultSet $textMatches
+	): void {
+		if ( !( $textMatches instanceof CirrusSearchResultSet ) ) {
+			return;
+		}
+
+		self::setCurrentQuery( $textMatches->getElasticaResultSet()->getQuery()->getQuery() );
+	}
+
+	private static function setCurrentQuery( AbstractQuery $query ): void {
+		$GLOBALS[WikibaseFacetedSearchExtension::QUERY_GLOBAL] = $query;
+	}
+
 	public static function onSpecialSearchResultsAppend(
 		SpecialSearch $specialSearch,
 		OutputPage $output,
 		string $term
 	): void {
 		$output->addHTML(
 			WikibaseFacetedSearchExtension::getInstance()->getSidebarHtmlBuilder( $specialSearch->getLanguage() )->createHtml(
-				searchQuery: $term
+				searchQuery: $term,
+				currentQuery: self::getCurrentQuery()
 			)
 		);
 	}
 
+	private static function getCurrentQuery(): AbstractQuery {
+		return $GLOBALS[WikibaseFacetedSearchExtension::QUERY_GLOBAL] ?? new MatchAll();
```

Comment:
In which cases does the global not get set?

Reply:
I removed it. This was more of a theoretical fallback, although I am not sure what would cause it. At least in our case, if any tab was selected that should always casue at least some query to exist.